platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java

```diff
@@ -25,6 +26,7 @@
  */
 @UiThread
 @SuppressLint("StaticFieldLeak")
+@Keep
```

Comment:
Keeping just the `getAssetManager` method that is reference from JNI is enough in this context and would allow obfuscating all other java methods.

Reply:
we also look up the class from JNI, so it needs to be kept as well

Reply:
I believe that if you keep at least one method from the class, the class name is not going to be obfuscated. This allows all other methods that are not accessed from native to be obfuscated though.

Reply:
verified in https://github.com/mapbox/mapbox-gl-native/blob/201a53386f9f51cba2113b0a58541370d76c64af/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java that this doesn't work.

Reply:
Huh, interesting. It fails because if we're looking up a static method, JNI tries to verify the instance first with:
`Caused by: java.lang.NoSuchMethodError: no static method "Lcom/mapbox/mapboxsdk/Mapbox;.hasInstance()Z"`

The class is not obfuscated, but this internal, static `hasInstance()` method seems to be? Anyway :green_apple:

Reply:
we probably could fine tune this with using a specific proguard rule..